pkg/operator/configobserver/cloudconfig/sync_cloud_config.go

```diff
@@ -0,0 +1,122 @@
+package cloudprovider
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	corelisterv1 "k8s.io/client-go/listers/core/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+)
+
+const (
+	configNamespace                 = "openshift-config"
+	machineSpecifiedConfigNamespace = "openshift-config-managed"
+	machineSpecifiedConfig          = "kube-cloud-config"
+)
+
+// InfrastructureLister lists infrastrucre information and allows resources to be synced
+type InfrastructureLister interface {
+	InfrastructureLister() configlistersv1.InfrastructureLister
+	ResourceSyncer() resourcesynccontroller.ResourceSyncer
+	ConfigMapLister() corelisterv1.ConfigMapLister
+}
+
+// NewCloudConfigSyncer returns a new cloudprovider dependent observer for syncing cloud config per provider
+// for controller-manager and api-server.
+func NewCloudConfigSyncer(targetNamespaceName string) configobserver.ObserveConfigFunc {
+	cloudObserver := &cloudConfigSyncer{
+		targetNamespaceName: targetNamespaceName,
+	}
+	return cloudObserver.SyncConfigMap
+}
+
+type cloudConfigSyncer struct {
+	targetNamespaceName string
+}
+
+// ObserveCloudProviderNames observes the cloud provider from the global cluster infrastructure resource.
+func (c *cloudConfigSyncer) SyncConfigMap(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(InfrastructureLister)
+	var errs []error
```

Comment:
get rid of this, you are not using it :-)

Reply:
Right, I would love to, but this method should be one of the config observers and implement [this](https://github.com/openshift/library-go/blob/2df205bdd55006a63d3fd22099db73432a0f9bd8/pkg/operator/configobserver/config_observer_controller.go#L42). I refactored the code so this will be more obvious when the error is returned.